packages/perseus-editor/src/components/locked-figures-section.tsx

```diff
@@ -25,9 +25,12 @@ type Props = {
 const LockedFiguresSection = (props: Props) => {
     // Keep track of all figures' accordions' expanded states for the
     // expand/collapse all button. Set the whole array to false initially.
-    const [expandedStates, setExpandedStates] = React.useState(
-        Array(props.figures?.length).fill(false),
-    );
+    const collapsedStateArray =
+        props.figures?.length && props.figures.length > 0
```

Comment:
Question: Wouldn't the first part of this check be sufficient? Undefined would be falsy, so if `figures` is undefined, this equals the empty array, but also if it is zero it is falsy. So if there is no `figures` defined or if `length` is zero, it's falsy. So the only way `props.figures?.length` can be truthy is if it exists and is greater than 0.

Reply:
That's what I thought! My guess is that it saw the length is not null, and so it evaluated to `true`? which also = 1?

Would love if someone with better knowledge of JavaScript had some insight into this. (Maybe @benchristel?)

Reply:
> Wouldn't the first part of this check be sufficient?

This is subjective, but I'm not a fan of using bare numbers for truthy checks (`if props.figures?.length`). I find them tricky to reason about. 

@Myranae is right about `0` being falsey though.

<img width="206" alt="image" src="https://github.com/Khan/perseus/assets/77138/a9be7fb7-adbe-40ed-8237-c7ffc70bfe6d">


I _think_ I'd re-arrange this slightly and avoid needing to know the details of what is and isn't falsy/truthy in JS. 

```js
    const collapsedStateArray =
        Array((props.figures ?? []).length)
            .fill(false);
```

Reply:
@jeremywiebe So if it's an empty array that you try to fill with `false`, it will just stay an empty array, right?

Reply:
Correct.  `array.fill` with one argument replaces all instances in the array with the provided argument.  

<img width="259" alt="image" src="https://github.com/Khan/perseus/assets/77138/1eafa516-c1c5-40dc-bcff-c27558dd67a4">

Reply:
ah, that's much more elegant. thank you!